Core/GDCore/Project/ObjectConfiguration.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+#ifndef GDCORE_OBJECTCONFIGURATION_H
+#define GDCORE_OBJECTCONFIGURATION_H
+#include "GDCore/Vector2.h"
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "GDCore/Project/Behavior.h"
+#include "GDCore/Project/EffectsContainer.h"
+#include "GDCore/Project/VariablesContainer.h"
+#include "GDCore/String.h"
+#include "GDCore/Tools/MakeUnique.h"
+namespace gd {
+class PropertyDescriptor;
+class Project;
+class Layout;
+class ArbitraryResourceWorker;
+class InitialInstance;
+class SerializerElement;
+class EffectsContainer;
+}  // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Base class used to represent an object configuration
```

Comment:
```suggestion
 * \brief Base class used to represent an object configuration.
 * For example, this can be the animations in a sprite, the text, its font, its color in a Text object, etc...
```